src/main/java/org/verdictdb/coordinator/ScramblingCoordinator.java

```diff
@@ -40,7 +33,13 @@
 import org.verdictdb.exception.VerdictDBValueException;
 import org.verdictdb.sqlwriter.QueryToSql;
 
-import com.google.common.base.Optional;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
 
 public class ScramblingCoordinator {
```

Comment:
The current logic first calculates the number of blocks from the original table size; then, reduce the number of blocks if the relative size is set to smaller than 1.0. This approach will be less optimal when the relative size is small (e.g., 0.01).

An alternative way we can do is
```java
int effectiveTableRowCount = tableRowCount * relativeSize;
int blockCount = min(maxScrambleTableBlockCount, ceil(effectiveTableRowCount / blockSize));
double blockSize = effectiveTableRowCount / (double) blockCount;
int blockCountUpperBound = ceil(tableRowCount / blockSize);
```
Then, use sql query as follows:
```sql
CREATE TABLE scramble AS
SELECT * FROM (
  SELECT *, floor(rand() * blockCountUpperBound) as verdictdbblock 
  FROM originalTable
) t
WHERE verdictdbblock < blockCount
```

Reply:
I guess your point is that the current approach may create a scramble table larger than specified size. However, in other words, the current approach guarantees at least 1% (or 1 million rows) of original data gets included in the scramble table with 'maxBlockCount=100', but I think your proposed approach has a potential problem of allowing a scramble table having too few rows? 

Think about an extreme case where `effectiveTableRowCount` becomes 1 with `relativeSize` that is really small (like 0.00000001). We will end up with one single-row block for the scramble table (assuming we also guard the case where blockSize becomes 0) that is not going to work for any approximation. I guess it comes down to a decision about how much we let users to control the size of their data with this command.  Wdyt?

Reply:
Hmm. Your example confuses me. If the user wants, say, 1e-6 fraction of the original rows to be included in a scramble, why should we stop it?

Let me give you an example to clarify what I meant. Suppose that the original table has 1B rows and the relative size is 0.01 (thus, the effective size is 1B*0.01=10M). Then, I think we should create 10 blocks (each with size 1M) rather than creating 1 block with 10M rows.

However, since the number of block count limit is 100, the current logic first set the block count to 100 where each block contains 1B/100 = 10M rows. Then, since the relative size is 0.01, only the first block (of size 10M) will be created.

Reply:
> say, 1e-6 fraction of the original rows to be included in a scramble, why should we stop it?
My question was about allowing/preventing users to create scrambles that may have no approximation power at all. Why do we allow users to create scrambles that will be useless anyway? I am just saying that there could be such case for the record. 

I will make the change anyway as I agree on the other point about having 1M * 10 blocks rather than 10M * 1 block.